### PR TITLE
Add Timeline component

### PR DESCRIPTION
**Pull request by lialila**
Preview: https://storybook-qxyg5vcl0-treely.vercel.app/?path=/story/slices-timeline--minimal

### CODE REVIEW

File: src/slices/Timeline/Timeline.stories.tsx [hunk 1]
```diff
@@ -0,0 +1,220 @@
...
+    title: 'Title',
+    subTitle: 'Sub title',
+    timelineItems: [
+      {
+        ...timelineItem,
+        button: {
+          id: 1,
+          text: 'Button',
+          url: 'https://tree.ly',
+        },
```
Comment (lukasbals) [inline src/slices/Timeline/Timeline.stories.tsx:86]: You could creata a reusable button like the timelineItem

File: src/slices/Timeline/Timeline.stories.tsx [hunk 2]
```diff
@@ -0,0 +1,220 @@
...
+};
+
+export const WithBadgeOrLogo = Template.bind({});
+WithBadgeOrLogo.args = {
+  slice: {
+    tagline: 'Tagline',
+    title: 'Title',
+    subTitle: 'Sub title',
+    timelineItems: [
+      { ...timelineItem, badge: { text: 'Badge', variant: 'green' } },
```
Comment (lukasbals) [inline src/slices/Timeline/Timeline.stories.tsx:131]: You could also create a reusable badge

File: src/slices/Timeline/Timeline.stories.tsx [hunk 3]
```diff
@@ -0,0 +1,220 @@
...
+export const WithImage = Template.bind({});
+WithImage.args = {
+  slice: {
+    tagline: 'Tagline',
+    title: 'Title',
+    subTitle: 'Sub title',
+    timelineItems: [
+      {
+        ...timelineItem,
+        image: { id: 71, alt: 'Alt', img: { data: storybookStrapiCoverMock } },
```
Comment (lukasbals) [inline src/slices/Timeline/Timeline.stories.tsx:157]: Same for the image

File: src/slices/Timeline/Timeline.stories.tsx [hunk 4]
```diff
@@ -0,0 +1,220 @@
...
+    title: 'Title',
+    subTitle: 'Sub title',
+    timelineItems: [
+      {
+        ...timelineItem,
+        icon: {
+          id: 1,
+          alt: 'Icon alt text',
+          img: { data: storybookStrapiTreeIconMock },
+        },
```
Comment (lukasbals) [inline src/slices/Timeline/Timeline.stories.tsx:192]: Same

File: src/slices/Timeline/Timeline.tsx [hunk 5]
```diff
@@ -0,0 +1,321 @@
...
+import { CDN_URI } from '../../constants/api';
+import { IntlContext } from 'react-intl';
+import { BREAKPOINT_MD_QUERY } from '../../constants/breakpoints';
+import { useState, useEffect } from 'react';
+import { FormattedMessage } from 'react-intl';
+
+interface TimelineProps {
+  slice: {
+    title: string;
+    subTitle?: string;
```
Comment (lukasbals) [inline src/slices/Timeline/Timeline.tsx:33]: I think it's usually called text and not subTitle, right? 

I would keep consistence.

File: src/slices/Timeline/Timeline.tsx [hunk 6]
```diff
@@ -0,0 +1,321 @@
...
+  const { formatMessage } = useContext(IntlContext);
+  const [visibleItems, setVisibleItems] = useState(3);
+  const [mobile] = useMediaQuery(BREAKPOINT_MD_QUERY);
+
+  const showMoreItems = () => {
+    setVisibleItems((prevVisibleItems) => prevVisibleItems + 3);
+  };
+
+  return (
+    <>
```
Comment (lukasbals) [inline src/slices/Timeline/Timeline.tsx:63]: Not needed

File: src/slices/Timeline/Timeline.tsx [hunk 7]
```diff
@@ -0,0 +1,321 @@
...
+                                    {item.tagline}
+                                  </Text>
+                                  <Spacer h="2" />
+                                </>
+                              )}
+                              <Heading size="lg">{item.title}</Heading>
+                            </Box>
+                            {item.badge && (
+                              <Tag
+                                colorScheme={`${item.badge.variant}`}
```
Comment (lukasbals) [inline src/slices/Timeline/Timeline.tsx:219]:
```suggestion
                                colorScheme={item.badge.variant}
```

Doesn't this work?

**Review by lukasbals — COMMENTED** [review]
Good work!

File: src/slices/Timeline/Timeline.tsx (hunk 5 above)
Comment (lialila) [reply src/slices/Timeline/Timeline.tsx:33]: it is called `text` in `DefaultSectionHeader`, but in slice we always call it `subTitle`

Comment (lukasbals) [reply src/slices/Timeline/Timeline.tsx:33]: Really? Even if it's a rich text? Ok

Comment (lialila) [reply src/slices/Timeline/Timeline.tsx:33]: no, here you are right, `richText` is text. I'm fixing it

File: src/slices/Timeline/Timeline.tsx [hunk 8]
```diff
@@ -0,0 +1,329 @@
...
+                        </Box>
+                        {item.badge && (
+                          <Tag
+                            colorScheme={item.badge.variant}
+                            borderRadius="md"
+                          >
+                            {item.badge.text}
+                          </Tag>
+                        )}
+                        {item.logo && (
```
Comment (rschiffer) [inline src/slices/Timeline/Timeline.tsx:219]: What happens if a `badge` and a `logo` is defined?

File: package-lock.json
Comment (lukasbals) [inline package-lock.json:1]: Is this change needed? Since you didn't change the `package.json` file this file should not be changed.

**Review by lukasbals — COMMENTED** [review]
Otherwise, LGTM

File: src/slices/Timeline/Timeline.test.tsx [hunk 9]
```diff
@@ -0,0 +1,202 @@
...
+describe('The Timeline component', () => {
+  afterEach(() => {
+    pushSpy.mockRestore();
+  });
+
+  afterAll(() => {
+    useRouter.mockRestore();
+  });
+
+  it('displays the correct tagline and text, if they are defined in slice', () => {
```
Comment (rschiffer) [inline src/slices/Timeline/Timeline.test.tsx:36]:
```suggestion
  it('displays the correct tagline and text, if they are defined in the slice', () => {
```

File: src/slices/Timeline/Timeline.test.tsx [hunk 10]
```diff
@@ -0,0 +1,202 @@
...
+        ...defaultProps.slice,
+        tagline: 'Timeline tagline',
+        text: 'Timeline text',
+      },
+    });
+    expect(screen.getByText('Timeline tagline')).toBeInTheDocument();
+    expect(screen.getByText('Timeline text')).toBeInTheDocument();
+  });
+
+  it('displays the timline items', () => {
```
Comment (rschiffer) [inline src/slices/Timeline/Timeline.test.tsx:48]:
```suggestion
  it('displays the timeline items', () => {
```

File: src/slices/Timeline/Timeline.test.tsx [hunk 11]
```diff
@@ -0,0 +1,202 @@
...
+          { id: 2, title: 'Item 2' },
+        ],
+      },
+    });
+
+    expect(screen.getByText('Item 1')).toBeInTheDocument();
+    expect(screen.getByText('Item 2')).toBeInTheDocument();
+  });
+
+  it('displays the tagline and text in the timline items, if they are defined in slice', () => {
```
Comment (rschiffer) [inline src/slices/Timeline/Timeline.test.tsx:63]:
```suggestion
  it('displays the tagline and text in the timeline items, if they are defined in the slice', () => {
```

File: src/slices/Timeline/Timeline.test.tsx [hunk 12]
```diff
@@ -0,0 +1,202 @@
...
+          },
+        ],
+      },
+    });
+
+    expect(screen.getByText('Item 1 tagline')).toBeInTheDocument();
+    expect(screen.getByText('Item 1 text')).toBeInTheDocument();
+  });
+
+  it('displays the button in the timline items, if it is defined in slice', () => {
```
Comment (rschiffer) [inline src/slices/Timeline/Timeline.test.tsx:81]:
```suggestion
  it('displays the button in the timeline items, if it is defined in the slice', () => {
```

File: src/slices/Timeline/Timeline.test.tsx [hunk 13]
```diff
@@ -0,0 +1,202 @@
...
+          },
+        ],
+      },
+    });
+
+    expect(screen.getByRole('link')).toBeInTheDocument();
+    expect(screen.getByText('Optional Button')).toBeInTheDocument();
+  });
+
+  it('displays the badge if it is defined in slice', () => {
```
Comment (rschiffer) [inline src/slices/Timeline/Timeline.test.tsx:102]:
```suggestion
  it('displays the badge if it is defined in the slice', () => {
```

File: src/slices/Timeline/Timeline.test.tsx [hunk 14]
```diff
@@ -0,0 +1,202 @@
...
+            badge: { text: 'Badge', variant: 'green' },
+          },
+        ],
+      },
+    });
+
+    expect(screen.getAllByText('Badge')[0]).toBeInTheDocument();
+  });
+
+  it('displays the logo, icon and image if they are defined in slice', () => {
```
Comment (rschiffer) [inline src/slices/Timeline/Timeline.test.tsx:118]:
```suggestion
  it('displays the logo, icon and image if they are defined in the slice', () => {
```

File: src/slices/Timeline/Timeline.test.tsx [hunk 15]
```diff
@@ -0,0 +1,202 @@
...
+        ],
+      },
+    });
+
+    expect(
+      screen.getByAltText(messagesEn['sections.timeline.backgroundShapes'])
+    ).toBeInTheDocument();
+  });
+
+  it('displays the "Show three more milestones" button if more than three timeline items defined in slice', () => {
```
Comment (rschiffer) [inline src/slices/Timeline/Timeline.test.tsx:174]:
```suggestion
  it('displays the "Show three more milestones" button if more than three timeline items are defined in the slice', () => {
```

File: src/slices/Timeline/Timeline.test.tsx [hunk 16]
```diff
@@ -0,0 +1,202 @@
...
+};
+
+describe('The Timeline component', () => {
+  afterEach(() => {
+    pushSpy.mockRestore();
+  });
+
+  afterAll(() => {
+    useRouter.mockRestore();
+  });
```
Comment (rschiffer) [inline src/slices/Timeline/Timeline.test.tsx:34]: I don't think you are using these mocks right?

File: src/slices/Timeline/Timeline.test.tsx
Comment (rschiffer) [inline src/slices/Timeline/Timeline.test.tsx:1]: Would it make sense to add a test case where we simulate the click on the button to load more items?

**Review by rschiffer — COMMENTED** [review]
Looks awesome 🤩

File: src/slices/Timeline/Timeline.test.tsx
Comment (lialila) [reply src/slices/Timeline/Timeline.test.tsx:1]: cool, will do it!

File: src/slices/Timeline/Timeline.tsx (hunk 8 above)
Comment (lialila) [reply src/slices/Timeline/Timeline.tsx:219]: i fixed them so they will be displayed in a row close to each other on the right side of the timeline card. But this shouldn't happen, because in the prototype was mentioned to display the `logo` or `badge`, not the both at the same time.

Comment (lialila) [reply src/slices/Timeline/Timeline.tsx:219]: @rschiffer do you think I should include this case it in Storybook?

**Review by lukasbals — COMMENTED** [review]
LGTM
But you need a commit message that triggers a release. Check out the readme.

File: src/slices/Timeline/Timeline.tsx (hunk 8 above)
Comment (rschiffer) [reply src/slices/Timeline/Timeline.tsx]: No, I think it's fine then.

**Review by lukasbals — APPROVED** [review]
